src/main/java/mate/academy/springbookapp/repository/shoppingcart/ShoppingCartRepository.java

```diff
@@ -1,12 +1,15 @@
 package mate.academy.springbookapp.repository.shoppingcart;
 
 import mate.academy.springbookapp.model.ShoppingCart;
-import org.springframework.data.jpa.repository.EntityGraph;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.stereotype.Repository;
 
 @Repository
 public interface ShoppingCartRepository extends JpaRepository<ShoppingCart, Long> {
-    @EntityGraph(attributePaths = {"cartItems", "user"})
+    @Query("select sc from ShoppingCart sc "
+            + "left join fetch sc.cartItems "
+            + "join fetch sc.user u "
+            + "where u.id = :userId")
```

Comment:
You should be able to handle these queries with `@EntityGraph` by adding necessary references, e.g. `"cartItem.book"`.
You might test it if you want or stick with `@Query`.